packages/alchemy/src/middleware/gas-fees.ts

```diff
@@ -1,22 +1,28 @@
-import { applyFeeOption, type BigNumberish } from "@alchemy/aa-core";
+import {
+  applyFeeOption,
+  isBigNumberish,
+  isPercentage,
+  type BigNumberish,
+} from "@alchemy/aa-core";
 import type { AlchemyProvider } from "../provider.js";
 import type { ClientWithAlchemyMethods } from "./client.js";
 
 export const withAlchemyGasFeeEstimator = (
   provider: AlchemyProvider
 ): AlchemyProvider => {
   provider.withFeeDataGetter(async (struct, overrides, feeOptions) => {
-    const maxPriorityFeePerGas =
-      overrides?.maxPriorityFeePerGas != null
-        ? overrides?.maxPriorityFeePerGas
-        : // it's a fair assumption that if someone is using this Alchemy Middleware, then they are using Alchemy RPC
-          applyFeeOption(
-            await (provider.rpcClient as ClientWithAlchemyMethods).request({
-              method: "rundler_maxPriorityFeePerGas",
-              params: [],
-            }),
-            feeOptions?.maxPriorityFeePerGas
-          );
+    const maxPriorityFeePerGas = isBigNumberish(overrides?.maxPriorityFeePerGas)
+      ? overrides?.maxPriorityFeePerGas!
+      : // it's a fair assumption that if someone is using this Alchemy Middleware, then they are using Alchemy RPC
+        applyFeeOption(
+          await (provider.rpcClient as ClientWithAlchemyMethods).request({
+            method: "rundler_maxPriorityFeePerGas",
+            params: [],
+          }),
+          isPercentage(overrides?.maxPriorityFeePerGas)
+            ? overrides?.maxPriorityFeePerGas
+            : feeOptions?.maxPriorityFeePerGas
+        );
```

Comment:
can we not use the `applyOverride` method to handle this logic here?

Reply:
hmm okay. but any critical reason?

Reply:
updated